### DISCUSSION TITLE
Error: 403 Country, region, or territory not supported

### DISCUSSION THREAD

**Discussion by smukerji** [vercel/community#7451] [Help] [answered]
### Summary

We are running our app on vercel hosted in US region. We have some customers who is accessing the website from Hong Kong. Our application makes a call at the backend to Open AI API for text inferencing. Currently we are getting Error: 403 Country, region, or territory not supported, error because the edge server detects the location of the request is coming from Hong Kong. Can anyone help how to resolve this, as we have our customers using the application, and no one from HK can have access to the application.

### Steps to Reproduce

1. Torri.AI 
2. click the chatbot
3. ask a question on the chatbot
4. the answer is blocked and no reply is retured. 
5. checking the application logs
6. Error: 403 Country, region, or territory not supported comes out at the backend.

**Comment by amyegan** [comment, marked as answer]
Hey @smukerji. OpenAI does not support the Hong Kong region (hkg1).

> You can prevent this change affecting your deployments by specifying allowed regions for your functions and [excluding Hong Kong](https://vercel.com/docs/functions/configuring-functions/region#setting-regions-in-your-function). Changing regions requires a redeployment of your application.

https://vercel.com/changelog/openai-will-not-support-the-hong-kong-region-hkg1-for-functions

**Reply by enricoros** [reply]
@amyegan @smukerji have you figured out how to exclude the region?
I see the link to the documentation, but there's no explicit instruction, and trying out all the combinations seems like a risky and defect-prone move.
Would appreciate official instructions on how to just exclude that one region, without impacting the order/operations in the other regions.

**Reply by amyegan** [reply]
You can specify allowed regions for a function as shown here: https://vercel.com/docs/functions/configuring-functions/region#setting-regions-in-your-function

The full region list can be found here: https://vercel.com/docs/edge-network/regions#region-list

**Reply by enricoros** [reply]
Thanks for the prompt answer @amyegan . Quick questions that probably many will have:

- I have many edge functions - do I need to do this procedure to them all? I would go the vercel.json route, but it doesn't seem to support the `preferredRegion` setting (there's a `region` setting, but seems to be for serverless and not edge functions)

- would `preferredRegion` alter the order of regions in addition to disallowing regions, e.g. would european customers be routed to pacific edge functions if that's the order of the preferred regions?

- assuming I take the full list and remove `hkg1`, how do we keep the list updated with vercel's upstream if there are regions added or removed in the future?

Thanks in advance.

**Reply by zineanteoh** [reply]
> I have many edge functions - do I need to do this procedure to them all? I would go the vercel.json route, but it doesn't seem to support the preferredRegion setting (there's a region setting, but seems to be for serverless and not edge functions)

Have the same concerns + questions as @enricoros. How do we share `preferredRegion` across multiple edge functions? 

## Approach 1 (failed) 
Tried to first have all our edge functions import the variable from another file: 
```ts
// src/common/llm.ts
export const PREFERRED_REGION = [
  'arn1',
  'bom1',
  'cdg1',
  'cle1',
  'cpt1',
  'dub1',
  'fra1',
  'gru1',
  /* "hkg1", Disabling hongkong edge runtime */
  'hnd1',
  'iad1',
  'icn1',
  'kix1',
  'lhr1',
  'pdx1',
  'sfo1',
  'sin1',
  'syd1',
];

// /api/llm/chat.ts
export const preferredRegion = PREFERRED_REGION;
```

But Vercel deployment throws the following error: 
<img width="600" alt="image" src="https://github.com/user-attachments/assets/47562b4a-557a-4bc5-bb3e-6b7fa676cebe">

## Approach 2 (failed)
Tried to use an alternative syntax: 
```ts
export { PREFERRED_REGIONS as preferredRegion } from '@/common/llm'; 
```

Vercel deployment passes but the regions default to "global" (not respecting the regions provided). 

<img width="600" src="https://github.com/user-attachments/assets/4762a01a-5d34-42a2-affd-2f3433caf06e" />

[a comment by an automated account is not shown]
